libs/core/mappers/based-params.mapper.ts

```diff
@@ -0,0 +1,39 @@
+import { AnimeParamsDto } from '../dtos/based-params.dto';
+import { AnimeType } from '../models/anime-type';
+import { AnimeParams } from '../models/based-params';
+
+import { AnimeTypeMapper } from './anime-type.mapper';
+
+export namespace AnimeParamsMapper {
+
+	/**
+	 * Converts a string array to AnimeType array.
+	 *
+	 * @param array The string array to convert.
+	 */
+	export function toArrayAnimeType(array: string[]): AnimeType[] {
+		const animeTypeValues = Object.values(AnimeType);
+		const isValid = array.every(value => animeTypeValues.includes(value as AnimeType));
+		return isValid ? array as AnimeType[] : AnimeParams.defaultValues.filterByType;
+	}
+
+	/**
+	 * Maps model to dto.
+	 * @param model AnimeParams model.
+	 */
+	export function toDto(model: AnimeParams): AnimeParamsDto {
+		const typeFilter = Array.isArray(model.filterByType) ?
+			model.filterByType.map(AnimeTypeMapper.toDto).join(',') :
```

Comment:
```suggestion
			model.filterByType.map((type) => AnimeTypeMapper.toDto(type)).join(',') :
```
Be careful, in this case you have lost the context